app/models/katello/erratum.rb

```diff
@@ -46,10 +46,8 @@ def self.repository_association_class
     end
 
     def self.applicable_to_hosts(hosts)
-      self.joins(:content_facet_errata).joins(:content_facets).
-         where("#{Katello::Host::ContentFacet.table_name}.host_id" => hosts).
-         select("DISTINCT ON (#{self.table_name}.updated, #{self.table_name}.id) #{self.table_name}.*").
-         order("#{self.table_name}.updated desc, #{self.table_name}.id")
+      self.joins(:content_facets).
+        where("#{Katello::Host::ContentFacet.table_name}.host_id" => hosts).uniq
```

Comment:
Yeah, i don't know that this is easily testable via a unit test.  As beav said the results from the method were correct, just slow.  Ideally we'd have some sort of automated tests tracking the performance of various pages and actions and that would have spotted the performance issue.